hack/runtime-migrator/cmd/backup/backup.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	gardener_types "github.com/gardener/gardener/pkg/client/core/clientset/versioned/typed/core/v1beta1"
+	"github.com/kyma-project/infrastructure-manager/hack/runtime-migrator-app/internal/backup"
+	"github.com/kyma-project/infrastructure-manager/hack/runtime-migrator-app/internal/config"
+	"github.com/kyma-project/infrastructure-manager/hack/runtime-migrator-app/internal/shoot"
+	"github.com/kyma-project/infrastructure-manager/pkg/gardener/kubeconfig"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"log/slog"
+	"time"
+)
+
+const (
+	timeoutK8sOperation = 20 * time.Second
+	expirationTime      = 60 * time.Minute
+)
+
+type Backup struct {
+	shootClient        gardener_types.ShootInterface
+	kubeconfigProvider kubeconfig.Provider
+	outputWriter       backup.OutputWriter
+	results            backup.Results
+	cfg                config.Config
+}
+
+func NewBackup(cfg config.Config, kubeconfigProvider kubeconfig.Provider, shootClient gardener_types.ShootInterface) (Backup, error) {
+	outputWriter, err := backup.NewOutputWriter(cfg.OutputPath)
+	if err != nil {
+		return Backup{}, err
+	}
+
+	return Backup{
+		shootClient:        shootClient,
+		kubeconfigProvider: kubeconfigProvider,
+		outputWriter:       outputWriter,
+		results:            backup.NewBackupResults(outputWriter.NewResultsDir),
+		cfg:                cfg,
+	}, nil
+}
+
+func (b Backup) Do(ctx context.Context, runtimeIDs []string) error {
+	listCtx, cancel := context.WithTimeout(ctx, timeoutK8sOperation)
+	defer cancel()
+
+	shootList, err := b.shootClient.List(listCtx, v1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	backuper := backup.NewBackuper(b.cfg.IsDryRun, b.kubeconfigProvider)
+
+	for _, runtimeID := range runtimeIDs {
+		shootToBackup, err := shoot.Fetch(ctx, shootList, b.shootClient, runtimeID)
+		if err != nil {
+			errMsg := fmt.Sprintf("Failed to fetch shoot: %v", err)
+			b.results.ErrorOccurred(runtimeID, "", errMsg)
+			slog.Error(errMsg, "runtimeID", runtimeID)
+
+			continue
+		}
+
+		if shoot.IsBeingDeleted(shootToBackup) {
+			continue
+		}
+
+		runtimeBackup, err := backuper.Do(ctx, *shootToBackup)
+		if err != nil {
+			errMsg := fmt.Sprintf("Failed to backup runtime: %v", err)
+			b.results.ErrorOccurred(runtimeID, shootToBackup.Name, errMsg)
+			slog.Error(errMsg, "runtimeID", runtimeID)
+
+			continue
+		}
+
+		if b.cfg.IsDryRun {
```

Comment:
Please consider refactoring this part by reducing `if-else` just to `if`.
This could improve code readability by removing nested code sections.